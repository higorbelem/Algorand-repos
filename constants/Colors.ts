const tintColorDark = '#fff';

export const Colors = {
  text: '#FFFFFF',
  background: '#191C21',
  tint: tintColorDark,
  icon: '#9BA1A6',
  tabIconDefault: '#9BA1A6',
  tabIconSelected: tintColorDark,
  main: '#2D2DF1',
  green: '#238636'
};
