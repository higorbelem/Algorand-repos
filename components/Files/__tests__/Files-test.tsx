import * as React from 'react';
import { render } from '@testing-library/react-native';

import { reposContentMock } from '@/constants/mocks';

import { Files } from '../index';

const mockData = reposContentMock.slice(0, 10);

describe('Files component', () => {
  it('Renders correctly', () => {
    const tree = render(<Files data={mockData} onItemSelected={() => {}}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show correct number of items', async () => {
    const tree = render(<Files data={mockData} onItemSelected={() => {}}/>);

    const components = await tree.findAllByTestId('files-item');

    expect(components).toHaveLength(mockData.length);
  });
});