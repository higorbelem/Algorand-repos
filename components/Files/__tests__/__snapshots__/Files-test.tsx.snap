// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Files component Renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "borderColor": "#FFFFFF",
      "borderRadius": 20,
      "borderWidth": 1,
      "justifyContent": "space-between",
    }
  }
  showsVerticalScrollIndicator={false}
  style={
    {
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 0,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-folder-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        .ci
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-folder-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        .github
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-file-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        .gitignore
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-file-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        LICENSE
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-file-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        README.md
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-file-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        build.gradle
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-folder-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        gradle
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-file-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        gradlew
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-file-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        gradlew.bat
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderTopColor": "#FFFFFF",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "gap": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="files-item"
    >
      <svg
        height={20}
        testID="files-item-folder-icon"
        width={20}
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        lib
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
